### ISSUE TITLE
Error when using GLFW =< 3.3.x on 1.16.5 Instance

### ISSUE THREAD

**Issue by Samauroh** [PrismLauncher/PrismLauncher#1838]
### Operating System

macOS

### Version of Prism Launcher

Prism Launcher 8.0
Build date: 2023-11-07

### Version of Qt

Qt version 6.6.0.

### Description of bug

[13:34:27] [Render thread/WARN] [minecraft/Main]: Failed to create window: 
net.minecraft.client.MainWindow$GlException: **GLFW error 65548: Cocoa: Regular windows do not have icons on macOS**
	at net.minecraft.client.MainWindow.func_208034_b(MainWindow.java:280) ~[?:?] {re:classloading,xf:OptiFine:default}
	at org.lwjgl.glfw.GLFWErrorCallbackI.callback(GLFWErrorCallbackI.java:43) ~[lwjgl-glfw-3.3.2.jar:build 13] {}
	at org.lwjgl.system.JNI.invokePPV(Native Method) ~[lwjgl-3.3.2.jar:build 13] {}
	at org.lwjgl.glfw.GLFW.nglfwSetWindowIcon(GLFW.java:2368) ~[lwjgl-glfw-3.3.2.jar:build 13] {}
	at org.lwjgl.glfw.GLFW.glfwSetWindowIcon(GLFW.java:2400) ~[lwjgl-glfw-3.3.2.jar:build 13] {}
	at net.minecraft.client.MainWindow.func_216529_a(MainWindow.java:226) ~[?:?] {re:classloading,xf:OptiFine:default}
	at net.minecraft.client.Minecraft.<init>(Minecraft.java:424) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
	at net.minecraft.client.main.Main.main(Main.java:149) ~[minecraft-1.16.5-client.jar:?] {re:classloading,pl:runtimedistcleaner:A}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_311] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_311] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_311] {}
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_311] {}
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) ~[forge-1.16.5-36.2.34-launcher.jar:36.2] {}
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.1.3.jar:?] {}
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.1.3.jar:?] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_311] {}
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_311] {}
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_311] {}
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_311] {}
	at io.github.zekerzhayard.forgewrapper.installer.Main.main(Main.java:66) [ForgeWrapper-1.5.7-prism.jar:1.5.7-prism] {}
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:87) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:130) [NewLaunch.jar:?] {}
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:70) [NewLaunch.jar:?] {}

### Steps to reproduce

On an 1.16.5 Minecraft Instance, modify the GLWF/LWJGL Version to be above 3.3

### Suspected cause

GLWF/LWJGL Implemented a new feature that is not being handled correctly/checked.

(https://github.com/PrismLauncher/PrismLauncher/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml#issuecomment-967169933)
This change in behavior was introduced with [glfw commit](https://github.com/glfw/glfw/commit/9a87c2a4b49808d3666c7b093b77e8512fa23908). When Minecraft is updated to use LWJGL 3.3.0, they'll have to either ignore the **GLFW_FEATURE_UNAVAILABLE** error or stop calling icon functions on macOS.

### This issue is unique

- [X] I have searched the issue tracker and did not find an issue describing my bug.

**Comment by Scrumplex** [comment]
Are you using a custom GLFW version? If so make sure to build off of glfw master. LWJGL does not use the stable 3.3.x branch

**Comment by Samauroh** [comment]
I meant =< 3.0On Nov 12, 2023, at 5:10 PM, Sefa Eyeoglu ***@***.***> wrote:﻿
Are you using a custom GLFW version? If so make sure to build off of glfw master. LWJGL does not use the stable 3.3.x branch

**Comment by Samauroh** [comment]
notice the log, the version is 3.3.2, however it is problematic with =<3.0 from my experieice

**Comment by Scrumplex** [comment]
I still need more details here.

- Are you using a custom version of glfw?
- If yes, what version is glfw (it is NOT 3.3.2, that's just LWJGL's version)

**Comment by Samauroh** [comment]
I got it. I modified and compiled the latest GLFW- my version nulls put cocoa/macOS window Icon functions. From here, I used a native version of Java 8 via Zulu and was able to launch with no problem. A bit of a manual fix, but I am running a Modpack that includes Pixelmon 9.1.7 on Minecraft 1.16.5 entirely natively (ARM64).Went from 15-30fps to 65-120fpsOn Nov 12, 2023, at 5:10 PM, Sefa Eyeoglu ***@***.***> wrote:﻿
Are you using a custom GLFW version? If so make sure to build off of glfw master. LWJGL does not use the stable 3.3.x branch
